### PR TITLE
feat: convert ProtoCluster factories to single CalorimeterIslandCluster_factoryT

### PR DESCRIPTION
**Pull request by wdconinc**
### Briefly, what does this PR introduce?
This converts the numerous `ProtoCluster*` factories to a single CalorimeterIslandCluster_factoryT with config struct. Shouldn't introduce any changes to the output collections.

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [x] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [x] Changes have been communicated to collaborators: @veprbl 

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No.

### Does this PR change default behavior?
No.

### CODE REVIEW

File: src/detectors/BHCAL/BHCAL.cc [hunk 1]
```diff
@@ -52,14 +50,34 @@ extern "C" {
...
           "HcalBarrelTruthProtoClusters", {"HcalBarrelRecHits", "HcalBarrelHits"}, {"HcalBarrelTruthProtoClusters"},
           app   // TODO: Remove me once fixed
         ));
-        app->Add(new JChainFactoryGeneratorT<ProtoCluster_factory_HcalBarrelIslandProtoClusters>(
-	    {"HcalBarrelRecHits"}, "HcalBarrelIslandProtoClusters"
-	));
+        app->Add(new JChainMultifactoryGeneratorT<CalorimeterIslandCluster_factoryT>(
+          "HcalBarrelIslandProtoClusters", {"HcalBarrelRecHits"}, {"HcalBarrelIslandProtoClusters"},
+          {
+            .adjacencyMatrix =
```
Comment (veprbl) [inline src/detectors/BHCAL/BHCAL.cc:61]:
```suggestion
            // Magic constants:
            //  24 - number of sectors
            //  5  - number of towers per sector
            .adjacencyMatrix =
```

File: src/detectors/FHCAL/FHCAL.cc [hunk 2]
```diff
@@ -245,8 +261,43 @@ extern "C" {
...
-          {"LFHCALRecHits"}, "LFHCALIslandProtoClusters"
+
+        // Magic constants:
+        //  24 - number of sectors
+        //  5  - number of towers per sector
+        //  moduleIDx
+        //  moduleIDy
+        //  towerx
+        //  towery
+        //  rlayerz
```
Comment (veprbl) [inline src/detectors/FHCAL/FHCAL.cc:272]: Those comments must have been copied over from SciGlass. If I had to guess:
```suggestion
        //  54 - number of modules in a row/column
        //  2  - number of towers in a module
```
cc @FriederikeBock @steinber 
Makes me wonder why there is no 4 (if 8M modules are used).

Comment (wdconinc) [reply src/detectors/FHCAL/FHCAL.cc]: Suggestion to address any necessary logic changes in a separate PR :-)

**Review by veprbl — APPROVED** [review]
Changes to parameters look good, no changes in the output.

**Pull request merged**